OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -92,25 +92,19 @@ public class ProductionQueue : IResolveOrder, ITick, ITechTreeElement, INotifyOw
 
 		// A list of things we could possibly build
 		readonly Dictionary<ActorInfo, ProductionState> producible = new Dictionary<ActorInfo, ProductionState>();
-		readonly List<ProductionItem> queue = new List<ProductionItem>();
+		protected readonly List<ProductionItem> Queue = new List<ProductionItem>();
 		readonly IEnumerable<ActorInfo> allProducibles;
 		readonly IEnumerable<ActorInfo> buildableProducibles;
 
 		Production[] productionTraits;
 
 		// Will change if the owner changes
 		PowerManager playerPower;
-		PlayerResources playerResources;
+		protected PlayerResources playerResources;
 		protected DeveloperMode developerMode;
 
 		public Actor Actor { get { return self; } }
 
-		[Sync] public int QueueLength { get { return queue.Count; } }
-		[Sync] public int CurrentRemainingCost { get { return QueueLength == 0 ? 0 : queue[0].RemainingCost; } }
-		[Sync] public int CurrentRemainingTime { get { return QueueLength == 0 ? 0 : queue[0].RemainingTime; } }
-		[Sync] public int CurrentSlowdown { get { return QueueLength == 0 ? 0 : queue[0].Slowdown; } }
-		[Sync] public bool CurrentPaused { get { return QueueLength != 0 && queue[0].Paused; } }
-		[Sync] public bool CurrentDone { get { return QueueLength != 0 && queue[0].Done; } }
```

Comment:
Any reason why these were removed?

Reply:
Because they were not used, and wont work properly with the new approach. [0] is not always the Current one. Depends on usage

Reply:
The problem with removing them is they have the [Sync] attribute, I would have to ask @pchote  if this will break something.

Reply:
It won't